tskv/src/summary.rs

```diff
@@ -577,10 +583,25 @@ impl Summary {
                     &mut file_metas,
                     min_seq.copied(),
                 );
+
+                let last_seq = new_version.last_seq();
                 let flushed_mem_cahces = mem_caches.get(&tsf_id);
                 tsf.write()
                     .await
                     .new_version(new_version, flushed_mem_cahces);
+
+                if !self.opt.raft_mode {
+                    let (task, rx) = WalTask::new_clear_wal_entry(
+                        tsf.read().await.tenant_database().to_string(),
+                        tsf_id,
+                        last_seq,
+                    );
+
+                    if self.wal_sender.send(task).await.is_ok() {
+                        let _ = rx.await;
```

Comment:
Execution of WalTask::ClearWalEntry opens all WAL files and read the footer from them, I think `rx.await` needs to be out of the loop:
```rs
receivers[]
the loop {
    ( tx, rx ) = get channel
    send wal task ( tx )
    receivers.push(rx)
}
for rx in receivers {
    let_ = rx.await
}
```